src/sonic-yang-models/yang-models/sonic-dtel.yang

```diff
@@ -0,0 +1,294 @@
+                    }
+                    default "false";
+                }
+
+                leaf drop_report {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+
+                leaf postcard {
+                    type string {
+                        pattern "true|false";
+                    }
+                    default "false";
+                }
+
+            }
+        }
+
+        container DTEL_REPORT_SESSION {
+            description "DTEL report session part of config_db.json";
+
+            list DTEL_REPORT_SESSION_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf src_ip {
+                    type inet:ip-address;
+                    description "DTEL report source IP address";
+                }
+
+                leaf-list dst_ip_list {
+                    type inet:ip-address;
+                    description "DTEL report destination IP addresses";
+                }
+
+                leaf truncate_size {
+                    type uint32;
+                    description "DTEL report truncate size";
+                }
+
+                leaf udp_dest_port {
+                    type string {
+                        pattern "0|[1-9][0-9]{0,3}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5]";
+                    }
+                    description "DTEL report UDP destination port";
+                }
+
+                leaf vrf {
+                    type union {
+                        type string {
+                            pattern "default";
+                        }
+                        type leafref {
+                            path "/vrf:sonic-vrf/vrf:VRF/vrf:VRF_LIST/vrf:name";
+                        }
+                    }
+                    description "DTEL report virtual router name";
+                }
+            }
+        }
+
+        container DTEL_QUEUE_REPORT {
+            description "DTEL_QUEUE_REPORT of config_db.json";
+
+            list DTEL_QUEUE_REPORT_LIST {
+                key "port qindex" ;
+
+                leaf port {
+                    type union {
+                        type leafref {
+                            path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                        }
+                        type leafref {
+                            path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
+                        }
+                    }
+                }
+
+                leaf qindex {
+                    type string {
+                        pattern "[0-7]?" {
+                            error-message "Invalid queue index";
+                            error-app-tag queue-index-invalid;
+                        }
```

Comment:
Is it possible to have more than 8 queues?

Reply:
It's depended on defferent hardwares.
Change type to uint8.